core/src/engine/coro/pool.hpp

```diff
@@ -119,31 +136,53 @@ typename Pool<Task>::CoroutinePtr Pool<Task>::GetCoroutine() {
       return *this;
     }
   };
-
   std::optional<Coroutine> coroutine;
   CoroutineMover mover{coroutine};
 
   // First try to dequeue from 'working set': if we can get a coroutine
   // from there we are happy, because we saved on minor-page-faulting (thus
   // increasing resident memory usage) a not-yet-de-virtualized coroutine stack.
-  if (used_coroutines_.try_dequeue(
-          GetUsedPoolToken<moodycamel::ConsumerToken>(), mover) ||
-      initial_coroutines_.try_dequeue(mover)) {
-    --idle_coroutines_num_;
+  if (!local_coro_buffer_.empty() || TryPopulateLocalCache()) {
+    coroutine = std::move(local_coro_buffer_.back());
+    local_coro_buffer_.pop_back();
+  } else if (initial_coroutines_.try_dequeue(mover)) {
+    idle_coroutines_num_.fetch_sub(1, std::memory_order_release);
   } else {
     coroutine.emplace(CreateCoroutine());
   }
+
   return CoroutinePtr(std::move(*coroutine), *this);
 }
 
 template <typename Task>
 void Pool<Task>::PutCoroutine(CoroutinePtr&& coroutine_ptr) {
-  if (idle_coroutines_num_.load() >= config_.max_size) return;
-  auto& token = GetUsedPoolToken<moodycamel::ProducerToken>();
-  const bool ok =
-      // We only ever return coroutines into our 'working set'.
-      used_coroutines_.enqueue(token, std::move(coroutine_ptr.Get()));
-  if (ok) ++idle_coroutines_num_;
+  local_coro_buffer_.push_back(std::move(coroutine_ptr.Get()));
+  if (local_coro_buffer_.size() <= kLocalCoroutineCacheMaxSize) {
+    return;
+  }
+
+  const std::size_t current_idle_coroutines_num =
+      idle_coroutines_num_.load(std::memory_order_acquire);
```

Comment:
release-acquire is not required here, but on x86*, `acquire` and `seq_cst` are the same. Let's remove the explicit memory order not to raise questions